### PR TITLE
 Implement getting detections from pipelines

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Fix several issues with data models and consumers thereof (mostly related to properly handling updates). Create new `NoaaPipelineWorker` class to extract detections from a pipeline, producing a dynamically-updating track model for pipelines which have detection output ports. Modify `noaa::gui::Window` to consume these track models.

### CODE REVIEW

File: sealtk/core/KwiverPipelinePortSet.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+}
+
+// ----------------------------------------------------------------------------
+void KwiverPipelinePortSet::bind(
+  kwiver::embedded_pipeline& pipeline, int index, PortType type,
+  std::vector<PortReference> additionalPorts)
+{
+  auto const timePortName = this->portName("timestamp", index);
```
Comment (KyleFromKitware) [inline sealtk/core/KwiverPipelinePortSet.cpp]: Can we expect that all pipelines will have a `timestamp` port?

**Comment by KyleFromKitware** [comment]
Please add a test case for `KwiverPipelinePortSet`.

File: sealtk/core/KwiverPipelinePortSet.cpp (hunk 1 above)
Comment (mwoehlke-kitware) [reply sealtk/core/KwiverPipelinePortSet.cpp]: That is an assumption we are making, yes.

**Comment by mwoehlke-kitware** [comment]
> Please add a test case for `KwiverPipelinePortSet`.

How? Suggestions? It's not something that can be readily tested without actually running a pipeline.

**Comment by KyleFromKitware** [comment]
You can create a dummy pipeline that writes to a `sink` port. (I'm not sure what to do about a read port.)

**Pull request merged**